changelog.d/12843.bugfix

```diff
@@ -0,0 +1 @@
+Fix ambiguous column name that would prevent use of MSC2716 History Import when using Postgres as a database.
```

Comment:
Can you rephrase this as what the end user would see, rather than what is happening internally? I.e. something like "Fix bug where servers using SQLite3 would fail to backfill in MSC2716 rooms" or something.

The changelog is aimed at server admins rather than developers

Reply:
I think the problem is that I have never used this feature and don't really understand how this would look to a server admin :-).

> Fix bug where servers using a Postgres database would fail to import historical messages using MSC2716.

Would that be more suitable?

Reply:
Yeah, potentially? @MadLittleMods may have a better idea

Reply:
It doesn't fail to import for me with my local Postgres Synapse setup.

Looking at this spot of code, I am guessing it will fail to backfill from another server when it sees an insertion event if you have `experimental_features.msc2716_enabled` enabled.

This seems more accurate:

> Fix bug where servers using a Postgres database would fail to backfill from an insertion event when [MSC2716](https://github.com/matrix-org/matrix-spec-proposals/pull/2716) is enabled (`experimental_features.msc2716_enabled`). 

---

```
$ psql -V
psql (PostgreSQL) 13.5

$ postgres -V
postgres (PostgreSQL) 13.5
```

Reply:
Thanks, my message was purely going off the name of the test. What you've written makes sense.